### PR TITLE
TST: disallow bare pytest raises

### PR DESCRIPTION
**Pull request by boweyism**
xref #30999 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Changes for:
-    pandas/tests/frame/methods/test_assign.py
-    pandas/tests/frame/methods/test_at_time.py
-    pandas/tests/frame/methods/test_between_time.py
-    pandas/tests/frame/methods/test_first_and_last.py
-    pandas/tests/frame/methods/test_interpolate.py
-    pandas/tests/frame/methods/test_replace.py
-    pandas/tests/frame/test_query_eval.py

Belongs to https://github.com/pandas-dev/pandas/issues/30999

### CODE REVIEW

File: pandas/tests/frame/test_query_eval.py [hunk 1]
```diff
@@ -870,7 +880,7 @@ def test_str_query_method(self, parser, engine):
 
             for lhs, op, rhs in zip(lhs, ops, rhs):
                 ex = f"{lhs} {op} {rhs}"
-                msg = r"'(Not)?In' nodes are not implemented"
+                # msg = r"'(Not)?In' nodes are not implemented"
```
Comment (simonjayhawkins) [inline pandas/tests/frame/test_query_eval.py:883]: don't leave this commented out. out of curiosity why move outside if block if message repeated on L911?

File: pandas/tests/frame/methods/test_assign.py [hunk 2]
```diff
@@ -65,9 +65,11 @@ def test_assign_bad(self):
         df = DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
 
         # non-keyword argument
-        with pytest.raises(TypeError):
+        msg = r"assign\(\) takes 1 positional argument but 2 were given"
+        with pytest.raises(TypeError, match=msg):
             df.assign(lambda x: x.A)
-        with pytest.raises(AttributeError):
+        msg = r"'DataFrame' object has no attribute 'C'"
```
Comment (simonjayhawkins) [inline pandas/tests/frame/methods/test_assign.py:71]: is the r prefix needed?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @boweyism for the PR. generally lgtm.

File: pandas/tests/frame/test_query_eval.py (hunk 1 above)
Comment (boweyism) [reply pandas/tests/frame/test_query_eval.py:883]: I left that in there by mistake. Also, I overlooked the if block. I'll put it there. (I had originally put it inside of the for loop)

Comment (simonjayhawkins) [reply pandas/tests/frame/test_query_eval.py:883]: can you remove this line entirely.

Comment (boweyism) [reply pandas/tests/frame/test_query_eval.py:883]: Yeah, I removed it:
https://github.com/pandas-dev/pandas/pull/34940/commits/14dd0b3c49ce161d4af237d6d40afee6775a538a

Comment (simonjayhawkins) [reply pandas/tests/frame/test_query_eval.py:883]: still showing in diff.  we don't want commented out code left behind.

Comment (boweyism) [reply pandas/tests/frame/test_query_eval.py]: whoops.. my bad. test_str_query_method and test_str_LIST_query_method got me confused. I've updated it

**Review by simonjayhawkins — APPROVED** [review]
Thanks @boweyism for the changes. lgtm pending green.

**Comment by jreback** [comment]
thanks @boweyism

**Pull request merged**